### PR TITLE
[typing] Add typing to the tests

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Refs #2079

### CODE REVIEW

File: tests/checkers/unittest_unicode/__init__.py [hunk 1]
```diff
@@ -82,5 +84,5 @@ class FakeNode:
     def __init__(self, content: bytes):
         self.content = io.BytesIO(content)
 
-    def stream(self):
+    def stream(self) -> BytesIO:
```
Comment (DanielNoord) [inline tests/checkers/unittest_unicode/__init__.py:87]: Doesn't `io.BytesIO` work?

File: tests/test_pylint_runners.py [hunk 2]
```diff
@@ -53,16 +54,16 @@ def test_pylint_run_jobs_equal_zero_dont_crash_with_cpu_fraction(
...
     builtin_open = open
 
-    def _mock_open(*args, **kwargs):
+    def _mock_open(*args: str, **kwargs) -> BufferedReader:
         if args[0] == "/sys/fs/cgroup/cpu/cpu.cfs_quota_us":
             return mock_open(read_data=b"-1")(*args, **kwargs)
         if args[0] == "/sys/fs/cgroup/cpu/cpu.shares":
             return mock_open(read_data=b"2")(*args, **kwargs)
-        return builtin_open(*args, **kwargs)
+        return builtin_open(*args, **kwargs)  # type: ignore[call-overload]
```
Comment (DanielNoord) [inline tests/test_pylint_runners.py:62]: isn't this a real issue?

File: tests/checkers/unittest_unicode/__init__.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply tests/checkers/unittest_unicode/__init__.py]: Nice catch, fixed !

File: tests/test_pylint_runners.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply tests/test_pylint_runners.py]: I added back ``typing.Any`` to fix this, in this case we really don't want to detail the full typing of ``builtin.open``.

File: tests/checkers/unittest_stdlib.py [hunk 3]
```diff
@@ -43,9 +42,9 @@ def test_deprecated_no_qname_on_unexpected_nodes(self) -> None:
         """
 
         def infer_func(
-            node: Name, context: Any | None = None  # pylint: disable=unused-argument
-        ) -> Iterator[Iterator | Iterator[AssignAttr]]:
-            new_node = nodes.AssignAttr(attrname="alpha", parent=node)
+            inner_node: nodes.Name, context=None  # pylint: disable=unused-argument
```
Comment (Pierre-Sassoulas) [inline tests/checkers/unittest_stdlib.py:45]:
```suggestion
            inner_node: nodes.Name, context: Any | None = None  # pylint: disable=unused-argument
```

File: tests/test_self.py [hunk 4]
```diff
@@ -74,10 +74,11 @@ def _configure_lc_ctype(lc_ctype: str) -> Iterator:
             os.environ[lc_ctype_env] = original_lctype
 
 
+# noinspection PyMissingConstructor
```
Comment (DanielNoord) [inline tests/test_self.py:77]: For which tool is this?

Comment (Pierre-Sassoulas) [reply tests/test_self.py:77]: Pycharm :)

Comment (DanielNoord) [reply tests/test_self.py:77]: Ah. Not sure we should add it. For gitignore we also leave out any IDE specific pragma's.

_There are also many things VS Code would like us to fix 😅_

Comment (Pierre-Sassoulas) [reply tests/test_self.py:77]:
```suggestion
```
Fair enough !

**Pull request merged**